### DISCUSSION TITLE
Fly shared IP doesn't seem to work

### DISCUSSION THREAD

**Discussion by RichardMore** [DigitallyRefined/docker-wireguard-tunnel#13] [Q&A]
Hi,

With the changes to Fly's pricing I switched over to shared IP with a new v3 deployment with the following:
```
fly launch
fly ips allocate-v4 --shared -a PROJECTNAME
```
During the new launch phase I chose the following answers:

> ? Would you like to copy its configuration to the new app? Yes
> ? Do you want to tweak these settings before proceeding? No
> ? Would you like to allocate a dedicated ipv4 address now? No

Is there anything that needs to be changed in order to make it work with the new share IP setup?

My fly.toml file is:
```
app = "PROJECTNAME"
primary_region = "ams"

[build]
  image = "ghcr.io/digitallyrefined/docker-wireguard-tunnel:v3"

[env]
  DOMAIN = "PROJECTNAME.fly.dev"
  PEERS = "1"
  SERVICES = "peer1:traefik:80:80,peer1:traefik:443:443"

[[mounts]]
  source = "wireguard_data"
  destination = "/etc/wireguard"
  auto_extend_size_threshold = 0

[[services]]
  protocol = "udp"
  internal_port = 51820

  [[services.ports]]
    port = 51820

[[services]]
  protocol = "tcp"
  internal_port = 80

  [[services.ports]]
    port = 80

[[services]]
  protocol = "tcp"
  internal_port = 443

  [[services.ports]]
    port = 443

[[vm]]
  cpu_kind = "shared"
  cpus = 1
  memory_mb = 1024
```

Tried IPv6 since that is dedicated but Fly doesn't support UDP forwarding over IPv6 yet.

**Comment by DigitallyRefined** [comment]
AFAIAA WireGuard requires a dedicated IPv4 to connect. It may also work over IPv6 if/when supported.

I did receive an email from Fly.io about upcoming changes to charge $2/month for each IPv4 address. However, I received another email from them saying that they wont charge for bills smaller than $5.00.

So as long as you keep within their [free allowance](https://fly.io/docs/about/pricing/#free-allowances), and have less that 3 IPv4 addresses then you shouldn't be charged 🤞
